plugins/inputs/dcos_containers/dcos_containers.go

```diff
@@ -0,0 +1,322 @@
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	"github.com/mesos/mesos-go/api/v1/lib"
+	"github.com/mesos/mesos-go/api/v1/lib/agent"
+	"github.com/mesos/mesos-go/api/v1/lib/agent/calls"
+	"github.com/mesos/mesos-go/api/v1/lib/httpcli"
+	"github.com/mesos/mesos-go/api/v1/lib/httpcli/httpagent"
+)
+
+const sampleConfig = `
+## The URL of the local mesos agent
+mesos_agent_url = "http://$NODE_PRIVATE_IP:5051"
+## The period after which requests to mesos agent should time out
+timeout = "10s"
+`
+
+// DCOSContainers describes the options available to this plugin
+type DCOSContainers struct {
+	MesosAgentUrl string
+	Timeout       internal.Duration
+}
+
+// measurement is a combination of fields and tags specific to those fields
+type measurement struct {
+	name   string
+	fields map[string]interface{}
+	tags   map[string]string
+}
+
+// combineTags combines this measurement's tags with some other tags. In the
+// event of a collision, this measurement's tags take priority.
+func (m *measurement) combineTags(newTags map[string]string) map[string]string {
+	results := make(map[string]string)
+	for k, v := range newTags {
+		results[k] = v
+	}
+	for k, v := range m.tags {
+		results[k] = v
+	}
+	return results
+}
+
+// newMeasurement is a convenience method for instantiating new measurements
+func newMeasurement(name string) measurement {
+	return measurement{
+		name:   name,
+		fields: make(map[string]interface{}),
+		tags:   make(map[string]string),
+	}
+}
+
+// SampleConfig returns the default configuration
+func (dc *DCOSContainers) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns a one-sentence description of dcos_containers
+func (dc *DCOSContainers) Description() string {
+	return "Plugin for monitoring mesos container resource consumption"
+}
+
+// Gather takes in an accumulator and adds the metrics that the plugin gathers.
+// It is invoked on a schedule (default every 10s) by the telegraf runtime.
+func (dc *DCOSContainers) Gather(acc telegraf.Accumulator) error {
+	uri := dc.MesosAgentUrl + "/api/v1"
+	cli := httpagent.NewSender(httpcli.New(httpcli.Endpoint(uri)).Send)
+	ctx, cancel := context.WithTimeout(context.Background(), dc.Timeout.Duration)
+	defer cancel()
+
+	gc, err := dc.getContainers(ctx, cli)
+	if err != nil {
+		return err
+	}
+
+	for _, c := range gc.Containers {
+		ts, tsOK := cTS(c)
+		tags := cTags(c)
+		for _, m := range cMeasurements(c) {
+			if len(m.fields) > 0 {
+				if tsOK {
+					acc.AddFields(m.name, m.fields, m.combineTags(tags), ts)
+				} else {
+					acc.AddFields(m.name, m.fields, m.combineTags(tags))
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+// getContainers requests a list of containers from the operator API
+func (dc *DCOSContainers) getContainers(ctx context.Context, cli calls.Sender) (*agent.Response_GetContainers, error) {
+	resp, err := cli.Send(ctx, calls.NonStreaming(calls.GetContainers()))
+	if err != nil {
+		return nil, err
+	}
```

Comment:
You could put 
`if err != nil {
            return nil, err
}`
on a separate function and call it on 108 and 112 lines.

Reply:
Wouldn't I have to test its response and return only if it was nil?

Reply:
Yep its true